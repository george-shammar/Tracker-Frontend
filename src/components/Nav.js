import React from 'react';
import { Link } from 'react-router-dom';
import '../stylesheets/nav.css';

const Nav = () => {
    return (
        <div className="nav">
          <Link to="/home">
          <div className="health-nav">
            <span className="material-icons logo-nav">maps_ugc</span>
            <p>Health<span className="p-subtitle">Track</span></p>
          </div>
          </Link>
          <Link to="/" className="log-out"><i class="far fa-user"></i></Link>
        </div>
    );
  }
  
  export default Nav;