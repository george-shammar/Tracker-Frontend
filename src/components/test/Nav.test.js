import React from 'react';
import ReactDom from 'react-dom';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import Nav from '../Nav';

it('renders Nav without crashing', () => {
    const div = document.createElement('div');
  
    ReactDom.render(
      <BrowserRouter>
        <Nav />
      </BrowserRouter>,
      div,
    );
  
    ReactDom.unmountComponentAtNode(div);
});

it('renders the Nav component correctly', () => {
    const tree = renderer.create(
      <BrowserRouter>
        <Nav />
      </BrowserRouter>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
});
