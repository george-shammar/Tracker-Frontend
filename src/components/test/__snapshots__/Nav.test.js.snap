// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the Nav component correctly 1`] = `
<div
  className="footer-nav"
>
  <div
    className="foot left"
  >
    <i
      className="fas fa-chart-line white"
    />
    <p>
      <a
        className="white font"
        href="/App"
        onClick={[Function]}
      >
        Healthtrack
      </a>
    </p>
  </div>
  <div
    className="foot"
  >
    <i
      className="far fa-chart-bar white center"
    />
    <p>
      <a
        className="white font"
        href="/Measurement"
        onClick={[Function]}
      >
        Add metrics
      </a>
    </p>
  </div>
  <div
    className="foot right"
  >
    <i
      className="far fa-user white"
    />
    <p>
      <a
        className="white font"
        href="/"
        onClick={[Function]}
      >
        Log Out
      </a>
    </p>
  </div>
</div>
`;
